gate-web/src/main/groovy/com/netflix/spinnaker/gate/health/DownstreamServicesHealthIndicator.groovy

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.gate.health
+
+import com.netflix.spinnaker.config.OkHttpClientConfiguration
+import com.netflix.spinnaker.gate.config.Service
+import com.netflix.spinnaker.gate.config.ServiceConfiguration
+import com.netflix.spinnaker.gate.services.internal.HealthCheckableService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.actuate.health.AbstractHealthIndicator
+import org.springframework.boot.actuate.health.Health
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import org.springframework.web.context.request.RequestContextHolder
+import org.springframework.web.context.request.ServletRequestAttributes
+import retrofit.RestAdapter
+import retrofit.RetrofitError
+import retrofit.client.OkClient
+
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+
+import static retrofit.Endpoints.newFixedEndpoint
+
+@Component
+class DownstreamServicesHealthIndicator extends AbstractHealthIndicator {
+  private final static Set<String> WHITELISTED_SERVICES = [
+    "orca", "clouddriver", "echo", "igor", "flex", "front50", "mahe"
```

Comment:
do we see this as a property in gate in case we add or remove new services or make them optional? i.e, fiat or igor